hubblestack/files/hubblestack_nova/misc.py

```diff
@@ -278,6 +278,210 @@ def test_failure_reason(reason):
+            return_list.append(output)
+            return_value = False
+    if return_value == False:
+        return str(return_list)
+    else:
+        return True
+
+def privileged_ports_not_mapped():
+    '''
+    Ensure that any privileged port ( below 1024 ) is not mapped within a docker container
+    '''
+    command = "docker ps --quiet | xargs docker inspect --format '{{ .Id }}: Ports={{ .NetworkSettings.Ports }}'"
+    output = _execute_shell_command(command)
+    ports = re.findall("HostPort:(\d+)]",output)
+    return_list = []
+    return_value = True
+    for port in ports:
+        if int(port) < 1024:
+            return_value  = False
+            return_list.append(port)
+
+    if return_value == False:
+        return return_list
+    else:
+        return True
+
+def ensure_sticky_bit_on_writable_directories():
+    '''
+    Ensure that sticky bit is set to all the world writable directories.
+    '''
+    command = "df --local -P | awk ' NR != 1 {printf $6}' | xargs -I '{}' find '{}' -xdev -type d \( -perm -0002 -a ! -perm -1000 \) 2>/dev/null"
+    output = _execute_shell_command(command)
+    if output.strip() == '':
+        return True
+    else:
+        return output
+
+def check_symlink(path,should_point_to):
+    '''
+    Ensure that the symlink is pointing to the path it should actually point to .
+    '''
+    command = 'find ' + path + ' -type l -ls'
+    output = _execute_shell_command( command )
+    if output.strip() == '':
+        return 'Not a symlink'
+    pointing_to = re.findall('->(.*)$',output)
+    if pointing_to[0].strip() == should_point_to.strip():
+        return True
+    else:
+        return (path + ' is incorrectly pointing to ' +  pointing_to)
+
+def check_time_synchronization():
+    '''
+    Ensure that some service is running to synchronize the system clock
+    '''
+    command = 'systemctl status systemd-timesyncd ntpd | grep "Active: active (running)"'
+    output = _execute_shell_command( command )
+    if output.strip() == '':
+        #return the output of systemctl for logging
+        partial_command = 'systemctl status systemd-timesyncd ntpd '
+        partial_output = _execute_shell_command( partial_command)
+        return partial_output
+    else:
+        return True
+
+
+def restrict_permissions(path,permission):
+    '''
+    Ensure that the file permissions on path are equal or more strict than the  pemissions given in argument
+    '''
+    path_details = __salt__['file.stats'](path)
+    given_permission = path_details.get('mode')
+    given_permission = given_permission[-3:]
+    permission = str(permission)
+    max_permission = permission[-3:]
+    if (is_valid_permission(max_permission[0],given_permission[0]) and is_valid_permission(max_permission[1],given_permission[1]) and is_valid_permission(max_permission[2],given_permission[2])):
+        return True
+    return given_permission
+
+def is_valid_permission(max_permission,given_permission):
```

Comment:
Let's even plan to change the name of this function to more user friendly. Like _is_permission_in_limit or something like that.